Atomic/util.py

```diff
@@ -232,19 +232,22 @@ def skopeo_layers(image, args=[], layers=[]):
     :param layers: if set, specify what layers must be downloaded
     :return: Returns the temporary directory with the layers
     """
+    ret = None
     temp_dir = tempfile.mkdtemp()
     try:
         args = ['skopeo', 'layers'] + args + [image] + layers
         r = subp(args, cwd=temp_dir)
         if r.return_code != 0:
             check_v1_registry(image)
             raise ValueError("Unable to interact with this registry: {}".format(r.stderr))
+        ret, temp_dir = temp_dir, None
     except OSError:
```

Comment:
Split this into two lines and add a comment.
# Successfully created return_dir, clear temp_dir field so it will not get removed.